### PR TITLE
backport of https://github.com/btcsuite/btcd/pull/1538/commits/c617c3536ad4312e15d150865c3af2bed6ef14b6

### PR DESCRIPTION
**Pull request by johnsonjh**

### CODE REVIEW

File: netsync/manager.go [hunk 1]
```diff
@@ -1085,7 +1085,7 @@ func (sm *SyncManager) handleInvMsg(imsg *invMsg) {
 
 	// Ignore invs from peers that aren't the sync if we are not current.
 	// Helps prevent fetching a mass of orphans.
-	if peer != sm.syncPeer && !sm.current() {
+	if !sm.current() && peer != sm.syncPeer {
```
Comment (cjdelisle) [inline netsync/manager.go:1088]: Why?
current() does more work than `peer != sm.syncPeer`

Comment (johnsonjh) [reply netsync/manager.go:1088]: It was my assumption (and apparently someone elses as well, theres a near-identical stalled mainline PR) that it makes more sense to check if it's current vs. the selected syncmanager's syncpeer merely as an order of importance.  I didn't do any actual benchmarking, but can't imagine the additional work would be of large significance.

Comment (cjdelisle) [reply netsync/manager.go:1088]: If you revert this part I will merge the rest, normally you should also do the cheapest check first, not the most "important".

Comment (johnsonjh) [reply netsync/manager.go:1088]: Will do

**Comment by johnsonjh** [comment]
correction on the way with request change reverted.

**Pull request closed without merging**